Sources/Satori/RequestRetry/TransientErrorAdapter.swift

```diff
@@ -46,9 +51,16 @@ public class TransientErrorAdapter: TransientErrorProtocol {
             return false
         }
     }
-    
-    func sendAsync<T>(request: @escaping () async throws -> T) async throws -> T {
-        return try await request()
-    }
 }
 
+/// An http adapter to intercept errors and determine if they are transient or not.
+public class TransientErrorHttpAdapter: TransientErrorProtocol {
+    var handler: TransientErrorHandler {
```

Comment:
In both Nakama and Satori could we remove `TransientErrorHttpAdapter` and `TransientErrorProtocol` instead just pass the `TransientErrorHandler` directly to the client for the `RetryInvoker` to use?